runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -969,6 +970,235 @@ J9::TransformUtil::canFoldStaticFinalField(TR::Compilation *comp, TR::Node* node
+      return false;
+      }
+
+   int32_t cpIndex = symRef->getCPIndex();
+   TR_OpaqueClassBlock* declaringClass = symRef->getOwningMethod(comp)->getClassFromFieldOrStatic(comp, cpIndex);
+   int32_t fieldNameLen;
+   char* fieldName = symRef->getOwningMethod(comp)->fieldName(cpIndex, fieldNameLen, comp->trMemory(), stackAlloc);
+   int32_t fieldSigLength;
+   char* fieldSignature = symRef->getOwningMethod(comp)->staticSignatureChars(cpIndex, fieldSigLength);
+
+   if (opt->trace())
+      {
+      traceMsg(comp,
+              "Looking at static final field n%dn %.*s declared in class %p\n",
+              node->getGlobalIndex(), fieldNameLen, fieldName, declaringClass);
+      }
+   if (!varHandleOnly || isVarHandleFolding(comp, declaringClass, fieldSignature, fieldSigLength))
+      {
+      if (J9::TransformUtil::foldStaticFinalFieldAssumingProtection(comp, node))
+         {
+         // Add class to assumption table
+         comp->addClassForStaticFinalFieldModification(declaringClass);
+         // Insert osrFearPointHelper call
+         TR::TreeTop* helperTree = TR::TreeTop::create(comp,
+                                                    TR::Node::create(node,
+                                                                     TR::treetop,
+                                                                     1,
+                                                                     TR::Node::createOSRFearPointHelperCall(node)));
+         currentTree->insertBefore(helperTree);
+
+         if (opt->trace())
+            {
+            traceMsg(comp,
+                    "Static final field n%dn is folded with OSRFearPointHelper call tree n%dn  helper tree n%dn\n",
+                    node->getGlobalIndex(), currentTree->getNode()->getGlobalIndex(), helperTree->getNode()->getGlobalIndex());
+            }
+
+         TR::DebugCounter::prependDebugCounter(comp,
+                                            TR::DebugCounter::debugCounterName(comp,
+                                                                               "staticFinalFieldFolding/success/(field %.*s)/(%s %s)",
+                                                                               fieldNameLen,
+                                                                               fieldName,
+                                                                               comp->signature(),
+                                                                               comp->getHotnessName(comp->getMethodHotness())),
+                                                                               currentTree->getNextTreeTop());
+         return true;
+         }
+      }
+   else
+      {
+      TR::DebugCounter::prependDebugCounter(comp,
+                                            TR::DebugCounter::debugCounterName(comp,
+                                                                               "staticFinalFieldFolding/notWorthFolding/(field %.*s)/(%s %s)",
```

Comment:
is `notWorthFolding` really the right name?

Reply:
Renamed to `notFolded`